utils/zerovec/src/varzerovec/slice.rs

```diff
@@ -60,41 +61,136 @@ pub struct VarZeroSlice<T: ?Sized> {
 }
 
 impl<T: VarULE + ?Sized> VarZeroSlice<T> {
-    /// Obtain a [`VarZeroVecBorrowed`] borrowing from the internal buffer
+    /// Construct a new empty VarZeroSlice
+    pub fn new_empty() -> &'static Self {
+        let arr: &[u8] = &[];
+        unsafe { mem::transmute(arr) }
+    }
+    /// Obtain a [`VarZeroVecComponents`] borrowing from the internal buffer
     #[inline]
-    pub fn as_borrowed<'a>(&'a self) -> VarZeroVecBorrowed<'a, T> {
+    pub(crate) fn as_components<'a>(&'a self) -> VarZeroVecComponents<'a, T> {
```

Comment:
Thought: This is a hot function, and I'm unsure about the code size implications.  If everything gets super inlined, the `VarZeroVecComponents` need not ever actually exist, and overall code size might co down.  But if these functions are never inlined, then we can do more code sharing.

Reply:
According to CI:

Before:
```
-rwxr-xr-x 1 runner docker    59664 Dec 20 02:10 optim5.elf
```

After:
```
-rwxr-xr-x 1 runner docker    58528 Dec 20 03:16 optim5.elf
```

So this PR actually reduces code size overall. I'm happy to investigate future improvements later then.

Reply:
Yeah, it reduces codesize by sharing all the codepaths better I think